metadetect/lsst/configs.py

```diff
@@ -66,6 +66,10 @@ def _fill_config(config):
     _verify_psf_config(config['psf'])
     _verify_detect_config(config['detect'])
 
+    # the new task can't handle None for weight
+    if config['weight'] is None:
+        del config['weight']
```

Comment:
Please don't delete config fields. If you want it to handle None, set `optional=True` in its definition.

Reply:
To be clear, the default for weight is None, but in the config used for the task it is a class called WeightConfig

definition of WeightClass

https://github.com/esheldon/metadetect/blob/e3b495ba8e289ca1997563c514e55f03004721c7/metadetect/lsst/metadetect.py#L91

It is used here

https://github.com/esheldon/metadetect/blob/e3b495ba8e289ca1997563c514e55f03004721c7/metadetect/lsst/metadetect.py#L186

Reply:
How do we make that entire thing optional rather than  individual fields in WeightConfig?

Reply:
That's not a supported operation. I guess I don't fully understand the use case here - do you just want to not use any of the values there?

Reply:
The weight will be ignored if the measurement type is gauss. So we could set some defaults and let it be ignored.

Reply:
Yeah, ignoring is fine. And if you are deleting it only internal to `metadetect`, that'd also be fine.